source/common/grpc/async_client_impl.h

```diff
@@ -55,11 +55,15 @@ class CallbackHolder {
 public:
   typedef std::unique_ptr<RawAsyncStreamCallbacks> CallbackPtr;
   CallbackPtr& getHeldCallback() { return callback_; }
-  void setHeldCallback(CallbackPtr&& callback) { callback_ = std::move(callback); }
+  void setHeldCallback(CallbackPtr&& callback, RawAsyncClientSharedPtr client = nullptr) {
```

Comment:
It's great this idea works! Can we add another function like "setHeldClient()" to separate the purposes?

Alternatively, instead of changing the function here, we can keep a reference of the shared pointer as part of the derived class of `RawAsyncStreamCallbacks` in serverless-proxy. It also makes sense since everything tracked by that callback class need to outlive the stream.

Reply:
🤦 yeah... that makes more sense 😄 i'll shift over there and close this PR